packages/flutter/lib/src/widgets/container.dart

```diff
@@ -303,18 +303,19 @@ class Container extends StatelessWidget {
   final BoxConstraints constraints;
 
   /// Empty space to surround the [decoration] and [child].
-  final EdgeInsets margin;
+  final EdgeInsetsGeometry margin;
 
   /// The transformation matrix to apply before painting the container.
   final Matrix4 transform;
 
-  EdgeInsets get _paddingIncludingDecoration {
+  EdgeInsetsGeometry _getPaddingIncludingDecoration(BuildContext context) {
     if (decoration == null || decoration.padding == null)
       return padding;
-    final EdgeInsets decorationPadding = decoration.padding;
+    final EdgeInsetsGeometry decorationPadding = decoration.padding;
     if (padding == null)
       return decorationPadding;
-    return padding + decorationPadding;
+    final TextDirection textDirection = Directionality.of(context);
+    return padding.resolve(textDirection) + decorationPadding.resolve(textDirection);
```

Comment:
The problem with the other code path is that one or both of the paddings are null, so resolving them is messy. But yeah, I'm not super happy with how this function turned out.